### PR TITLE
fix: bugfix response's patch and patchType is nil, controller panic

### PR DESCRIPTION
**Pull request by CharlesQQ**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
when bugfix response's patch and patchType is nil, controller panic

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/1578

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
@CharlesQQ You might have to sign your commit according to guideline: https://github.com/karmada-io/karmada/pull/1584/checks?check_run_id=5847682139

File: pkg/resourceinterpreter/customizedinterpreter/webhook/resourceinterpretercontext.go [hunk 1]
```diff
@@ -106,8 +106,12 @@ func verifyResourceInterpreterContext(operation configv1alpha1.InterpreterOperat
 		if err != nil {
 			return nil, err
 		}
-		res.Patch = response.Patch
-		res.PatchType = *response.PatchType
+		if response.Patch != nil {
+			res.Patch = response.Patch
+		}
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/customizedinterpreter/webhook/resourceinterpretercontext.go:111]: This judgment feels unnecessary.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
In addition to the modification here, you need to modify one more, can you help update it:
```diff
--- a/pkg/resourceinterpreter/customizedinterpreter/customized.go
+++ b/pkg/resourceinterpreter/customizedinterpreter/customized.go
@@ -252,6 +252,9 @@ func (e *CustomizedInterpreter) callHook(ctx context.Context, hook configmanager
 func applyPatch(object *unstructured.Unstructured, patch []byte, patchType configv1alpha1.PatchType) (*unstructured.Unstructured, error) {
        switch patchType {
        case configv1alpha1.PatchTypeJSONPatch:
+               if len(patch) == 0 {
+                       return object, nil
+               }
                patchObj, err := jsonpatch.DecodePatch(patch)
                if err != nil {
                        return nil, err
```

File: pkg/resourceinterpreter/customizedinterpreter/webhook/resourceinterpretercontext.go (hunk 1 above)
Comment (CharlesQQ) [reply pkg/resourceinterpreter/customizedinterpreter/webhook/resourceinterpretercontext.go]: updated! @XiShanYongYe-Chang

**Review by XiShanYongYe-Chang — COMMENTED** [review]
thanks!

/lgtm

**Comment by XiShanYongYe-Chang** [comment]
Hi @CharlesQQ , please add `release note` in the pull request description.

**Comment by CharlesQQ** [comment]
Done! please check @XiShanYongYe-Chang

**Comment by CharlesQQ** [comment]
Done! is it correct? please check @XiShanYongYe-Chang

**Comment by CharlesQQ** [comment]
understand, is that ok? @XiShanYongYe-Chang

**Review by RainbowMango — APPROVED** [review]
/approve

**Comment by RainbowMango** [comment]
@CharlesQQ Can you help `cherry-pick` this patch to both `release-1.0` and `release-1.1` branches?
You can do that by:
```bash
$ hack/cherry_pick_pull.sh release-1.0 1584
$ hack/cherry_pick_pull.sh release-1.1 1584
```

**Comment by CharlesQQ** [comment]
@RainbowMango ok

**Pull request merged**